docs/lang/articles/basic/sparse.md

```diff
@@ -170,31 +172,49 @@ The bitmasked SNodes are like dense SNodes with auxiliary activity values.
 
 ### Dynamic SNode
 
-To support variable-length fields, Taichi provides dynamic SNodes. The code snippet below first creates a 5x1 dense block (line 2). Then each cell of the dense block contains a variable-length dynamic container (line 3). The maximum length of the dynamic container is 5. In the `make_lists()` function, you can use `ti.append()` to add a value to the end of a dynamic SNode. `x.parent()` is the same as `pixel`. The dense field `l` stores the length of each dynamic SNode.
+To support variable-length fields, Taichi provides dynamic SNodes.
+The dynamic SNode must have only one axis, and the axis must be the last axis.
+No other SNode can be placed under a dynamic SNode, that is, a dynamic SNode must be directly placed with a field.
+There must not be another SNode which axes include the same axis as the dynamic SNode in the path from the dynamic SNode to the root of the SNode tree.
```

Comment:
I think we have stated many laws of the dynamic SNode, but we haven't shown the readers what a dynamic SNode looks like. Maybe it's better to put this paragraph in later sections?